capa/main.py

```diff
@@ -189,57 +189,32 @@ def has_rule_with_namespace(rules, capabilities, rule_cat):
     return False
 
 
+def is_internal_rule(rule):
+    return rule.meta.get("namespace", "").startswith("internal/")
+
+
+def is_file_limitation_rule(rule):
+    return rule.meta.get("namespace", "") == "internal/limitation/file"
+
+
 def has_file_limitation(rules, capabilities, is_standalone=True):
-    file_limitations = {
-        # capa will likely detect installer specific functionality.
-        # this is probably not what the user wants.
-        "executable/installer": [
-            " This sample appears to be an installer.",
-            " ",
-            " capa cannot handle installers well. This means the results may be misleading or incomplete."
-            " You should try to understand the install mechanism and analyze created files with capa.",
-        ],
-        # capa won't detect much in .NET samples.
-        # it might match some file-level things.
-        # for consistency, bail on things that we don't support.
-        "runtime/dotnet": [
-            " This sample appears to be a .NET module.",
-            " ",
-            " .NET is a cross-platform framework for running managed applications.",
-            " capa cannot handle non-native files. This means that the results may be misleading or incomplete.",
-            " You may have to analyze the file manually, using a tool like the .NET decompiler dnSpy.",
-        ],
-        # capa will detect dozens of capabilities for AutoIt samples,
-        # but these are due to the AutoIt runtime, not the payload script.
-        # so, don't confuse the user with FP matches - bail instead
-        "compiler/autoit": [
-            " This sample appears to be compiled with AutoIt.",
-            " ",
-            " AutoIt is a freeware BASIC-like scripting language designed for automating the Windows GUI.",
-            " capa cannot handle AutoIt scripts. This means that the results will be misleading or incomplete.",
-            " You may have to analyze the file manually, using a tool like the AutoIt decompiler MyAut2Exe.",
-        ],
-        # capa won't detect much in packed samples
-        "anti-analysis/packer/": [
-            " This sample appears to be packed.",
-            " ",
-            " Packed samples have often been obfuscated to hide their logic.",
-            " capa cannot handle obfuscation well. This means the results may be misleading or incomplete.",
-            " If possible, you should try to unpack this input file before analyzing it with capa.",
-        ],
-    }
+    file_limitation_rules = list(filter(is_file_limitation_rule, rules.rules.values()))
```

Comment:
need to sort these by namespace depth so we can apply more specific limitations first, e.g.
`anti-analysis/packer/vmprotect` before `anti-analysis/packer/`

Reply:
right now, all the file limitation rules live at the same depth under the namespace `internal/limitation/file`.  i see your point about wanting to prioritize some rules over the others, but depth won't work as implemented today.

notably, a vmprotect rule living in `anti-analysis/packer/vmprotect` will not trigger a file limitation on its own - there must be a `match` reference to this rule from some other rule living within `internal/limitation/file` as well.

Reply:
i don't know how to specify the order of file limitations. should that be addressed in this PR or later on, as we figure out when this is important?

Reply:
naive and simple solution is to have a meta field specifying the "priority" and manually assign these when a rule needs to be prioritized.

Reply:
oh, i see, this is about listing the rules that match the given namespace?

we can't really guarantee that the file limitation rules will use namespace matches; however, since this seems common today, we could add some special case logic to detect and render this data.

Reply:
gotcha, we can update it later then, if necessary